src/models/envelope/envelope_file.rs

```diff
@@ -0,0 +1,143 @@
+use serde_derive::{Deserialize, Serialize};
+
+use crate::interchange::DataInterchange;
+use crate::Result;
+use crate::{crypto::Signature, interchange::Json};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct EnvelopeFile {
+    payload: String,
+    payload_type: String,
+    signatures: Vec<Signature>,
+}
+
+impl EnvelopeFile {
+    #[allow(dead_code)]
+    // TODO: remove #[allow(dead_code)] after metadata deploy
+    pub fn new(payload: String, payload_type: String, signatures: Vec<Signature>) -> Self {
+        Self {
+            payload,
+            payload_type,
+            signatures,
+        }
+    }
+
+    /// standard serialize for EnvelopeFile
+    #[allow(dead_code)]
+    // TODO: remove #[allow(dead_code)] after metadata deploy
+    pub fn to_bytes(&self) -> Result<Vec<u8>> {
+        Json::canonicalize(&Json::serialize(self)?)
+    }
+
+    /// standard deserialize for EnvelopeFile
+    #[allow(dead_code)]
+    // TODO: remove #[allow(dead_code)] after metadata deploy
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self> {
+        let ret: Self = serde_json::from_slice(bytes)?;
+        Ok(ret)
+    }
+
+    #[allow(dead_code)]
+    // TODO: remove #[allow(dead_code)] after metadata deploy
+    pub fn signatures(&self) -> &Vec<Signature> {
+        &self.signatures
+    }
+
+    #[allow(dead_code)]
+    // TODO: remove #[allow(dead_code)] after metadata deploy
+    pub fn payload(&self) -> &String {
+        &self.payload
+    }
+
+    #[allow(dead_code)]
+    // TODO: remove #[allow(dead_code)] after metadata deploy
+    pub fn payload_type(&self) -> &String {
+        &self.payload_type
+    }
+}
+
+#[cfg(test)]
+mod test_envelope_file {
+    use std::str;
+
+    use once_cell::sync::Lazy;
+
+    use super::EnvelopeFile;
+    // TODO: change to mock test use mockall
+    use crate::crypto::Signature;
+
+    pub struct EnvelopeFileTuple<'a> {
+        name: String,
+        payload: String,
+        payload_type: String,
+        signatures: Vec<Signature>,
+        packet: &'a str,
+    }
+
+    pub static SERIALIZE_DATAS: Lazy<Vec<EnvelopeFileTuple>> = Lazy::new(|| {
+        vec![
+            EnvelopeFileTuple {
+                name: "blank_test".to_string(),
+                payload: "114514".to_string(),
+                payload_type: "link".to_string(),
+                signatures: Vec::new(),
+                packet: "{\"payload\":\"114514\",\"payload_type\":\"link\",\"signatures\":[]}",
+            },
+            EnvelopeFileTuple {
+                name: "blank_test".to_string(),
+                payload: "in-toto-rs".to_string(),
+                payload_type: "https://in-toto.io/payloadment/v0.1".to_string(),
```

Comment:
very minor nit, insignificant to the actual change, but we may as well use statement instead of payloadment.

Reply:
> very minor nit, insignificant to the actual change, but we may as well use statement instead of payloadment.

Absolutely agree. Actually there is another class `Envelope` used to do `sign` and `verify` of `EnvelopeFile` data model, it deal with state model, and convert it to bytes when generating `EnvelopeFile`.

Reply:
Glad you agree! If you can amend that string in the payload_type, this should be good to go from me. :)

Reply:
+1 here